WebEngine/Dialog.cpp

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "WebEngine/Dialog.h"
+
+#include <QHostAddress>
+#include <QMargins>
+#include <QWebChannel>
+#include <QWebEnginePage>
+#include <QWebEngineProfile>
+#include <QWebSocketServer>
+#include <memory>
+
+#include "JsonTransport.h"
+
+namespace WebEngine {
+
+Dialog::Dialog(QWebEngineProfile* profile, QWidget* parent) : QDialog(parent), page_(profile) {
+  web_socket_server_.reset(
+      // NOLINTNEXTLINE
```

Comment:
What is this for?

Reply:
clang-tidy will (and should) warn about the "new". But in this case we need it.